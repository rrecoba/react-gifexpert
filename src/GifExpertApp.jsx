import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components/index';


export const GifExpertApp = () => {
  
    const [categories, setCategories] = useState(['One Punch']);
    
    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories([...categories, newCategory]);
    }

    return (
    <>   
        {/* titulo */ }
        <h1>GifExpertApp</h1>

        {/* Input */ }
        <AddCategory onNewCategory={ (newCat)=> onAddCategory(newCat) } />

        {/* Listado de Gif */ }
        { 
            categories.map( ( category ) => (
                <GifGrid 
                    key={ category } 
                    category={ category }
                />
            ))
        }
    </>
  )
}
