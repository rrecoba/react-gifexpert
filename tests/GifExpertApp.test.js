import { render, screen, fireEvent } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Pruebas en GifExpertApp', () => { 

    const newCategory = 'Saitama';

    test('should Add new categories', () => {
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox'); 
        const form = screen.getByRole('form'); 
  
        //ahora disparo los eventos para agregar 3 categorias nuevas
        fireEvent.input(input, { target: { value: newCategory } });
        fireEvent.submit(form); //agrega Saitama
  
        //Espero que se hallan agregado 3 categorÃ­as cuyos nombres aparecen en etiquetas h3
        expect(screen.getAllByRole('heading', { level: 1 }).length).toBe(3);
    });

     test('should match the snapshot', () => {
 
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
     });
 })