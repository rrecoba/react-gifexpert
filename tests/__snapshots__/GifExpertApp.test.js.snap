// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp should match the snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h1>
    One Punch
  </h1>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
